Item.cpp

```diff
@@ -0,0 +1,7 @@
+#include "Item.h"
+
+Item::Item ( const std::string& name_, int effect ) : name(name_), boost(effect) { }
+
+Item::~Item() { }
+
+void Item::use(Character*) { }
```

Comment:
Remove this default implementation.
- [ ] revise

Reply:
I had implemented the Item::use(Character*) in Item.cpp thinking it was causing vtables warnings.

The project compiles, links, runs without the implementation of void Item::use(Character*)  in Item.cpp

However commenting out the Item destructor in Item.cpp causes linker errors:

/tmp/Character-2d10d7.o: In function `Character::defend()':
Character.cpp:(.text+0x62a): undefined reference to `typeinfo for Item'
/tmp/Character-2d10d7.o: In function `Character::help(Character&)':
Character.cpp:(.text+0x7c1): undefined reference to `typeinfo for Item'
/tmp/AttackItem-25571b.o: In function `AttackItem::~AttackItem()':
AttackItem.cpp:(.text._ZN10AttackItemD2Ev[_ZN10AttackItemD2Ev]+0x11): undefined reference to `Item::~Item()'
/tmp/AttackItem-25571b.o:(.rodata+0x50): undefined reference to `typeinfo for Item'
/tmp/DefensiveItem-839137.o: In function `DefensiveItem::~DefensiveItem()':
DefensiveItem.cpp:(.text._ZN13DefensiveItemD2Ev[_ZN13DefensiveItemD2Ev]+0x11): undefined reference to `Item::~Item()'
/tmp/DefensiveItem-839137.o:(.rodata+0x50): undefined reference to `typeinfo for Item'
/tmp/HelpfulItem-51e61c.o: In function `HelpfulItem::~HelpfulItem()':
HelpfulItem.cpp:(.text._ZN11HelpfulItemD2Ev[_ZN11HelpfulItemD2Ev]+0x11): undefined reference to `Item::~Item()'
/tmp/HelpfulItem-51e61c.o:(.rodata+0x50): undefined reference to `typeinfo for Item'
/tmp/Item-d9f7b3.o: In function `Item::Item(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)':
Item.cpp:(.text+0xa): undefined reference to `vtable for Item'
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
exit status 1

Reply:
Compiles , links and runs fine now,  in Item.h the virtual destructor is defined but not implemented
the vtables warnings are gone.

Item.h
-------

virtual ~Item();  // semicolon terminated its defined but no curly braces { } means its not implemented here in Item.h


Item.cpp
---------

Item::~Item() { } // destructor implemented here curly braces { }

That's how I understand it/